File: Python/74. Search a 2D Matrix.py
class Solution:
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        if len(matrix) == 0:
            return False
        n, m = len(matrix), len(matrix[0])
        i, j = 0, m-1
        while i < n and j >= 0:
            if target == matrix[i][j]:
                return True
            elif target < matrix[i][j]:
                j -= 1
            elif target > matrix[i][j]:
                i += 1
        return False

File: C++/74. Search a 2D Matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        if(matrix.size() == 0)
            return false;
        int m = matrix.size(), n = matrix[0].size();
        int j = n-1, i = 0;
        while(i<m && j>=0)
        {
            if(matrix[i][j] == target)
                return true;
            else if(target > matrix[i][j])
                i++;
            else if(target < matrix[i][j])
                j--;
        }
        return false;
    }
};